src/js/actions/libraries.js

```diff
@@ -546,6 +555,50 @@ define(function (require, exports) {
     applyCharacterStyle.transfers = [layerActions.resetLayers];
 
     /**
+     * Applies the color the selected layers. It currently supports two types of layers:
+     *  - Text layer: will set layer's font color
+     *  - Vector layer: will set layer's fill color
+     *
+     * @param {Color} color
+     *
+     * @return {Promise}
+     */
+    var applyColor = function (color) {
+        var currentDocument = this.flux.store("application").getCurrentDocument(),
+            selectedLayers = currentDocument ? currentDocument.layers.selected : null;
+
+        if (!currentDocument || selectedLayers.isEmpty()) {
+            return Promise.resolve();
+        }
+
+        var textLayers = selectedLayers.filter(function (l) { return l.isTextLayer(); }),
+            vectorLayers = selectedLayers.filter(function (l) { return l.isVector(); });
+
+        if (textLayers.isEmpty() && vectorLayers.isEmpty()) {
+            return Promise.resolve();
+        }
+
+        // FIXME: Setting font color and fill color at the same time will result in two histroy states.
+        //        We should merge the two states when transaction is supported.
+        return Promise.bind(this)
+            .then(function () {
+                return textLayers.isEmpty() ? null :
+                    this.transfer(typeActions.setColor, currentDocument, textLayers, color, true, true);
+            })
+            .then(function () {
+                return vectorLayers.isEmpty() ? null :
+                    this.transfer(shapeActions.setFillColor, currentDocument, vectorLayers, 0, color,
+                        true, true, true);
+            })
+            .then(function () {
+                return this.transfer(layerActions.resetLayers, currentDocument, textLayers.concat(vectorLayers));
```

Comment:
Do you need to resetLayers for color changes? Make sure it's absolutely necessary